src/routes/courierSize.tsx

```diff
@@ -1,199 +1,199 @@
-import React, { useEffect, useState } from 'react';
-import { createFileRoute, Link } from '@tanstack/react-router';
-import { getFirestore, collection, onSnapshot, deleteDoc, query, where, getDocs } from 'firebase/firestore';
-import { initializeApp } from 'firebase/app';
-import Swal from 'sweetalert2';
-import firebaseConfig from '../firebaseConfig'; // Adjust the path if necessary
-import courierImage from '../assets/img/bg-img/courier.png';
-
-// Initialize Firebase and Firestore
-const app = initializeApp(firebaseConfig);
-const db = getFirestore(app);
-
-// Define a type for the vault data
-type VaultItem = {
-  id: string;
-  packsize?: string;
-  compartment?: number;
-  status?: string;
-};
-
-export const Route = createFileRoute('/courierSize')({
-  component: RouteComponent,
-});
-
-function RouteComponent() {
-  const [vaultData, setVaultData] = useState<VaultItem[]>([]);
-  const [error, setError] = useState<string | null>(null);
-  const [filterStatus, setFilterStatus] = useState<string | null>(null);
-
-  useEffect(() => {
-    console.log('Attempting to fetch data from Firestore...');
-    const vaultCollection = collection(db, 'vault');
-
-    const unsubscribe = onSnapshot(
-      vaultCollection,
-      (snapshot) => {
-        if (!snapshot.empty) {
-          const data = snapshot.docs.map((doc) => ({
-            id: doc.id,
-            ...doc.data(),
-          })) as VaultItem[];
-          console.log('Fetched data:', data);
-          setVaultData(data);
-          setError(null);
-        } else {
-          console.log('Vault collection is empty.');
-          setVaultData([]);
-        }
-      },
-      (error) => {
-        console.error('Error fetching data:', error);
-        setError(error.message);
-      }
-    );
-
-    return () => unsubscribe();
-  }, []);
```

Comment:
Nice integration with firebase, might be more organized in a custom hook though.